plugins/destination/snowflake/client/spec.go

```diff
@@ -9,10 +21,97 @@ const (
 )
 
 type Spec struct {
-	ConnectionString   string `json:"connection_string,omitempty"`
-	BatchSize          int    `json:"batch_size,omitempty"`
-	BatchSizeBytes     int    `json:"batch_size_bytes,omitempty"`
-	MigrateConcurrency int    `json:"migrate_concurrency,omitempty"`
+	// ConnectionString is a DSN the user can specify with all config options,
+	// but we also support each attribute individually. See [sf.ParseDSN].
+	ConnectionString string `json:"connection_string,omitempty"`
+
+	// Account is [sf.Config.Account].
+	Account string `json:"account,omitempty"`
+	// User is [sf.Config.User].
+	User string `json:"user,omitempty"`
+	// Password is [sf.Config.Password].
+	Password string `json:"password,omitempty"`
+	// Database is [sf.Config.Database].
+	Database string `json:"database,omitempty"`
+	// Schema is [sf.Config.Schema].
+	Schema string `json:"schema,omitempty"`
+	// Warehouse is [sf.Config.Warehouse].
+	Warehouse string `json:"warehouse,omitempty"`
+	// Role is [sf.Config.Role].
+	Role string `json:"role,omitempty"`
+	// Region is [sf.Config.Region].
+	Region string `json:"region,omitempty"`
+
+	// Params is [sf.Config.Params] - other arbitrary connection
+	// options.
+	Params map[string]*string `json:"params,omitempty"`
```

Comment:
Thanks for the PR @icio, looks great 🚀 

Do we need all these spec options to support key pair authentication? If not I would keep the minimal set of options needed for that support.

Every configuration item becomes a part of our public API which means we need to continuously support it, test it, document it and do a breaking change if it changes, so we usually don't add configuration items unless strictly needed.